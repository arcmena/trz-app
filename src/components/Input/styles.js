import styled from 'styled-components';

export const Container = styled.div`
    display: flex;
    align-items: center;

    width: ${(props) => (props.size ? props.size : '320px')};
    height: 60px;
    border: 2px solid rgba(48, 48, 48, 0.35);
    border-radius: 15px;

    padding: 8px 10px;

    input {
        border: none;
        font-family: 'Ubuntu';
        font-size: 20px;
        color: #383838;
    }
`;
