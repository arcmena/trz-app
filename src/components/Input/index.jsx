import React from 'react';

import { Container } from './styles';

export default ({ size }) => {
    return (
        <Container>
            <input type="text" size={size} />
        </Container>
    );
};
