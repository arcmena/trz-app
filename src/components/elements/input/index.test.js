import React from 'react';
import { shallow } from 'enzyme';
import Input from './';

const essentialProps = {
    type: 'text',
    id: 'button',
    name: 'namedbutton',
    placeholder: 'sometext',
    readOnly: true,
    value: 'sometest',
};

describe('<Input/>', () => {
    it('should render as expected', () => {
        const wrapper = shallow(<Input {...essentialProps} />);
        expect(wrapper).toMatchSnapshot();
    });
    it('should render the input', () => {
        const wrapper = shallow(<Input {...essentialProps} />);
        expect(wrapper.find('input').props()).toEqual({
            type: essentialProps.type,
            id: essentialProps.id,
            name: essentialProps.name,
            placeholder: essentialProps.placeholder,
            readOnly: essentialProps.readOnly,
            value: essentialProps.value,
        });
    });
});
