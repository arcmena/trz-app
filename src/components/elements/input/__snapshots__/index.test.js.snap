// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input/> should render as expected 1`] = `
<styled.div>
  <input
    id="button"
    name="namedbutton"
    placeholder="sometext"
    readOnly={true}
    type="text"
    value="sometest"
  />
</styled.div>
`;
